x-pack/plugins/security_solution/public/detections/components/take_action_dropdown/index.tsx

```diff
@@ -87,6 +87,12 @@ export const TakeActionDropdown = React.memo(
     );
     const isEvent = actionsData.eventKind === 'event';
 
+    const isAgentEndpoint = useMemo(() => ecsData?.agent?.type?.includes('endpoint'), [ecsData]);
+
+    const isEndpointEvent = isEvent && isAgentEndpoint;
```

Comment:
nit: This and the `disableEventFilterAction` could be memoized to avoid unnecessary rerenders.